File: ExpressionParser.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq.Expressions;
using System.Text;

namespace YourNamespace
{
    public class ExpressionParser
    {
        private readonly DataTable dataTable;

        public ExpressionParser()
        {
            this.dataTable = new DataTable();
        }

        public Expression Parse(string expression)
        {
            try
            {
                return new DataTable().Compute(expression, null) as double?;
            }
            catch
            {
                throw new ArgumentException("Invalid expression");
            }
        }

        public Func<double> Compile(Expression expression)
        {
            try
            {
                var lambda = Expression.Lambda<Func<double>>(expression);
                var compiled = lambda.Compile();
                return compiled;
            }
            catch
            {
                throw new ArgumentException("Invalid expression");
            }
        }
    }
}

File: java_build/sigmath/ExpressionParser.java
package sigmath;

public class ExpressionParser
{
	private final DataTable dataTable;

	public ExpressionParser(DataTable dataTable)
	{
		this.dataTable = new DataTable();
	}

	public final Expression Parse(String expression)
	{
		try
		{
			return Expression.Constant((new DataTable()).Compute(expression, null), Double.class);
		}
		catch (java.lang.Exception e)
		{
			throw new IllegalArgumentException("Invalid expression");
		}
	}

	public final tangible.Func0Param<Double> Compile(Expression expression)
	{
		try
		{
			var lambda = Expression.<tangible.Func0Param<Double>>Lambda(expression);
			var compiled = lambda.Compile();
			return compiled;
		}
		catch (java.lang.Exception e)
		{
			throw new IllegalArgumentException("Invalid expression");
		}
	}
}
